background.js

```diff
@@ -49,47 +91,45 @@ const waitAndClick = async (querySelector) => {
   if (await waitForElement(querySelector)) await clickElement(querySelector);
 };
 
-browser.pageAction.onClicked.addListener(async () => {
-  await browser.tabs.executeScript({
+browser.pageAction.onClicked.addListener(async (tab) => {
+  browser.tabs.executeScript({
     code: `
-    if (!document.querySelector(".caal-alerts")) {
-      alerts = document.createElement("div")
-      alerts.setAttribute("class", "caal-alerts");
-      alerts.setAttribute("style", "position:absolute;");
-      document.body.appendChild(alerts);
-    }
-  `,
+      if (!document.querySelector(".caal-loading")) {
+        const loading = document.createElement("div");
+        loading.textContent = "Analysing chess game...";
+        loading.setAttribute("class", "caal-loading");
+        document.body.appendChild(loading);
+      }
+    `,
   });
 
-  await waitAndClick('.share');
+  await setLoadingState(true);
 
-  await waitAndClick('.board-tab-item-underlined-component');
+  const [playerName] = await browser.tabs.executeScript({
+    code: `document.querySelector('[data-test-element="user-tagline-username"]').textContent;`,
+  });
 
-  if (await waitForElement("[name='pgn']")) {
-    const [pgn] = await browser.tabs.executeScript({
-      code: `document.querySelector("[name='pgn']").value;`,
-    });
+  const pgn = await getPGN(playerName, tab.url.split("?")[0]);
 
-    await waitAndClick("[data-cy='share-menu-close']");
+  await setLoadingState(false);
 
-    await browser.tabs.create({ url: 'https://lichess.org/paste' });
+  await browser.tabs.create({ url: "https://lichess.org/paste" });
```

Comment:
Minor:

What happens if you switch tabs while the script is executing?

I believe the tab ID should be used in any actions that are supposed to be performed on the tab. I assume that at the moment, the script assumes the user will not switch tabs because it executes the scripts in the default tab (I assume it is the active one).

It could be more robust if the code remembered the tab IDs it is supposed to be working with

Reply:
This feature will be added in 1.7 version. Tomorrow I will create issue related with that problem